app/classifier/frame-annotator.jsx

```diff
@@ -137,54 +67,34 @@ export default class FrameAnnotator extends React.Component {
       );
     }
 
-    const svgStyle = {};
-    if (type === 'image' && !this.props.loading) {
-      // Images are rendered again within the SVG itself.
-      // When cropped right next to the edge of the image,
-      // the original tag can show through, so fill the SVG to cover it.
-      svgStyle.background = 'black';
-
-      // Allow touch scrolling on subject for mobile and tablets
-      if (taskDescription) {
-        if ((taskDescription.type !== 'drawing') && (taskDescription.type !== 'crop')) {
-          svgStyle.pointerEvents = 'none';
-        }
-      }
-      if (this.props.panEnabled === true) {
-        svgStyle.pointerEvents = 'all';
-      }
-    }
-
-    const svgProps = {};
-
-    if (TaskComponent) {
-      ({BeforeSubject, InsideSubject, AfterSubject} = TaskComponent);
-    }
-
     const hookProps = {
       taskTypes: tasks,
       workflow: this.props.workflow,
       task: taskDescription,
       classification: this.props.classification,
       annotation: this.props.annotation,
       frame: this.props.frame,
-      scale: this.getScale(),
       naturalWidth: this.props.naturalWidth,
       naturalHeight: this.props.naturalHeight,
-      containerRect: this.getSizeRect(),
-      getEventOffset: this.getEventOffset,
       onChange: this.props.onChange,
-      preferences: this.props.preferences,
-      normalizeDifference: this.normalizeDifference,
-      getScreenCurrentTransformationMatrix: this.getScreenCurrentTransformationMatrix
+      preferences: this.props.preferences
     };
 
-    Object.keys(tasks).map((task) => {
-      const Component = tasks[task];
-      if (Component.getSVGProps) {
-        Object.assign(svgProps, Component.getSVGProps(hookProps));
-      }
-    });
+    let subjectChildren = [];
+    if (!!InsideSubject && !this.props.panEnabled) {
+      subjectChildren.push(<InsideSubject key="inside" {...hookProps} />);
+    }
+    const persistentHooks = Object
+      .keys(tasks)
+      .map((taskName) => {
+        const PersistInsideSubject = tasks[taskName].PersistInsideSubject;
+        if (PersistInsideSubject) {
+          return <PersistInsideSubject key={taskName} {...hookProps} />;
+        }
+        return null;
+      })
+      .filter(Boolean);
+    subjectChildren = subjectChildren.concat(persistentHooks);
 
```

Comment:
These hooks should be pushed down to the individual renderers, since `TaskComponent.InsideSubject` and `TaskComponent.PersistInsideSubject` might generate HTML or SVG.